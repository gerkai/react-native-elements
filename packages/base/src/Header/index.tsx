import { withTheme } from '../config';
import { Header, HeaderProps } from './Header';

export { Header };
export type { HeaderProps };
export default withTheme<HeaderProps, {}>(Header, 'Header');
