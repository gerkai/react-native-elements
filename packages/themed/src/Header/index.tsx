import { withTheme } from '../config';
import {
  Header,
  HeaderProps,
} from '@react-native-elements/base/dist/Header/Header';

export { Header };
export type { HeaderProps };
export default withTheme<HeaderProps>(Header, 'Header');
