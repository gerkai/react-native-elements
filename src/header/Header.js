import React from 'react';
import PropTypes from 'prop-types';
import { Platform, StatusBar, StyleSheet, View } from 'react-native';

import Text from '../text/Text';
import Icon from '../icons/Icon';
import { colors, ViewPropTypes, getStatusBarHeight } from '../config';
import renderNode from '../helpers/renderNode';
import nodeType from '../helpers/nodeType';

const ALIGN_STYLE = {
  left: 'flex-start',
  right: 'flex-end',
  center: 'center',
};

const Children = ({ style, placement, children }) => (
  <View style={[{ alignItems: ALIGN_STYLE[placement] }, style]}>
    {children == null || children === false
      ? null
      : children.text
        ? renderNode(Text, children.text, { numberOfLines: 1, ...children })
        : children.icon
          ? renderNode(Icon, {
              ...children,
              name: children.icon,
              containerStyle: [
                { alignItems: ALIGN_STYLE[placement] },
                children.containerStyle,
              ],
            })
          : renderNode(Text, children)}
  </View>
);

const Header = ({
  statusBarProps,
  leftComponent,
  centerComponent,
  rightComponent,
  leftContainerStyle,
  centerContainerStyle,
  rightContainerStyle,
  backgroundColor,
  containerStyle,
  placement,
  barStyle,
  children = [],
  ...attributes
}) => (
  <View
    {...attributes}
    style={[
      styles.container,
      backgroundColor && { backgroundColor },
      containerStyle,
    ]}
  >
    <StatusBar barStyle={barStyle} {...statusBarProps} />
    <Children
      style={[placement === 'center' && styles.rightLeftContainer, leftContainerStyle]}
      placement="left"
    >
      {(React.isValidElement(children) && children) ||
        children[0] ||
        leftComponent}
    </Children>
    <Children
      style={[styles.centerContainer, placement !== 'center' && {
        paddingHorizontal: Platform.OS === 'ios' ? 15 : 16,
      }, centerContainerStyle]}
      placement={placement}
    >
      {children[1] || centerComponent}
    </Children>
    <Children
      style={[placement === 'center' && styles.rightLeftContainer, rightContainerStyle]}
      placement="right"
    >
      {children[2] || rightComponent}
    </Children>
  </View>
);

Header.propTypes = {
  placement: PropTypes.oneOf(['left', 'center', 'right']),
  leftComponent: nodeType,
  centerComponent: nodeType,
  rightComponent: nodeType,
  leftContainerStyle: ViewPropTypes.style,
  centerContainerStyle: ViewPropTypes.style,
  rightContainerStyle: ViewPropTypes.style,
  backgroundColor: PropTypes.string,
  containerStyle: ViewPropTypes.style,
  statusBarProps: PropTypes.object,
  barStyle: PropTypes.oneOf(['default', 'light-content', 'dark-content']),
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
};

Header.defaultProps = {
  placement: 'center',
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: colors.primary,
    borderBottomColor: '#f2f2f2',
    borderBottomWidth: StyleSheet.hairlineWidth,
    paddingHorizontal: 10,
    paddingTop: getStatusBarHeight(),
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    height: (Platform.OS === 'ios' ? 44 : 56) + getStatusBarHeight(),
  },
  centerContainer: {
    flex: 3,
  },
  rightLeftContainer: {
    flex: 1,
  },
});

export default Header;
