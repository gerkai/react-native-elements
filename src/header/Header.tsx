import React from 'react';
import {
  Platform,
  StatusBar,
  StyleSheet,
  View,
  ImageBackground,
  TextStyle,
  StyleProp,
  TextProps,
  ViewProps,
  StatusBarProps,
  StatusBarStyle,
  ImageSourcePropType,
  ImageStyle,
  ViewStyle,
  FlexAlignType,
} from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import { withTheme } from '../config';
import { renderNode } from '../helpers';

import Text from '../text/Text';
import Icon, { IconObject } from '../icons/Icon';
import { Theme } from '../config/theme';

type Placement = 'left' | 'center' | 'right';

const ALIGN_STYLE: Record<Placement, FlexAlignType> = {
  left: 'flex-start',
  right: 'flex-end',
  center: 'center',
};

type HeaderChildrenProps = {
  placement: Placement;
  style: StyleProp<ViewStyle>;
  children: any;
};

const Children = ({ style, placement, children }: HeaderChildrenProps) => (
  <View
    style={StyleSheet.flatten([{ alignItems: ALIGN_STYLE[placement] }, style])}
  >
    {children == null || children === false
      ? null
      : children.text
      ? renderNode(Text, children.text, { numberOfLines: 1, ...children })
      : children.icon
      ? renderNode(Icon, {
          ...children,
          name: children.icon,
          containerStyle: StyleSheet.flatten([
            { alignItems: ALIGN_STYLE[placement] },
            children.containerStyle,
          ]),
        })
      : renderNode(Text, children)}
  </View>
);

interface HeaderIcon extends IconObject {
  icon?: string;
  text?: string;
  color?: string;
  style?: StyleProp<TextStyle>;
}

type HeaderSubComponent = React.ReactElement<{}> | TextProps | HeaderIcon;

export type HeaderProps = ViewProps & {
  ViewComponent?: typeof React.Component;
  linearGradientProps?: Object;
  statusBarProps?: StatusBarProps;
  barStyle?: StatusBarStyle;
  leftComponent?: HeaderSubComponent;
  centerComponent?: HeaderSubComponent;
  rightComponent?: HeaderSubComponent;
  backgroundColor?: string;
  backgroundImage?: ImageSourcePropType;
  backgroundImageStyle?: ImageStyle;
  placement?: 'left' | 'center' | 'right';
  containerStyle?: StyleProp<ViewStyle>;
  centerContainerStyle?: StyleProp<ViewStyle>;
  leftContainerStyle?: StyleProp<ViewStyle>;
  rightContainerStyle?: StyleProp<ViewStyle>;
  theme?: Theme;
  children?: JSX.Element[];
  elevated: boolean;
};

const Header: React.FunctionComponent<HeaderProps> = (props: HeaderProps) => {
  React.useEffect(() => {
    const { linearGradientProps, ViewComponent } = props;
    if (linearGradientProps && !ViewComponent) {
      console.error(
        "You need to pass a ViewComponent to use linearGradientProps !\nExample: ViewComponent={require('react-native-linear-gradient')}"
      );
    }
  });

  const {
    statusBarProps,
    leftComponent,
    centerComponent,
    rightComponent,
    leftContainerStyle,
    centerContainerStyle,
    rightContainerStyle,
    backgroundColor,
    backgroundImage,
    backgroundImageStyle,
    containerStyle,
    placement = 'center',
    barStyle,
    children = [],
    linearGradientProps,
    ViewComponent = linearGradientProps || !backgroundImage
      ? View
      : ImageBackground,
    theme,
    elevated,
    ...attributes
  } = props;

  return (
    <>
      <StatusBar
        barStyle={barStyle}
        translucent={true}
        backgroundColor={backgroundColor || theme.colors.primary}
        {...statusBarProps}
      />
      <ViewComponent
        testID="headerContainer"
        {...attributes}
        style={StyleSheet.flatten([
          {
            borderBottomColor: '#f2f2f2',
            borderBottomWidth: StyleSheet.hairlineWidth,
            paddingHorizontal: 10,
            paddingVertical: 10,
            backgroundColor: theme.colors.primary,
            flexDirection: 'row',
            justifyContent: 'space-between',
            alignItems: 'center',
          },
          backgroundColor && { backgroundColor },
          elevated && styles.elevatedHeader,
          containerStyle,
        ])}
        source={backgroundImage}
        imageStyle={backgroundImageStyle}
        {...linearGradientProps}
      >
        <SafeAreaView
          edges={['left', 'top', 'right']}
          style={styles.headerSafeView}
        >
          <Children
            style={StyleSheet.flatten([
              placement === 'center' && styles.rightLeftContainer,
              leftContainerStyle,
            ])}
            placement="left"
          >
            {(React.isValidElement(children) && children) ||
              children[0] ||
              leftComponent}
          </Children>
          <Children
            style={StyleSheet.flatten([
              styles.centerContainer,
              placement !== 'center' && {
                paddingHorizontal: Platform.select({
                  android: 16,
                  default: 15,
                }),
              },
              centerContainerStyle,
            ])}
            placement={placement}
          >
            {children[1] || centerComponent}
          </Children>

          <Children
            style={StyleSheet.flatten([
              placement === 'center' && styles.rightLeftContainer,
              rightContainerStyle,
            ])}
            placement="right"
          >
            {children[2] || rightComponent}
          </Children>
        </SafeAreaView>
      </ViewComponent>
    </>
  );
};

const styles = StyleSheet.create({
  headerSafeView: {
    width: '100%',
    flexDirection: 'row',
  },
  centerContainer: {
    flex: 3,
  },
  rightLeftContainer: {
    flex: 1,
  },
  elevatedHeader: {
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 6,
    },
    shadowOpacity: 0.6,
    shadowRadius: 8.0,
    elevation: 24,
  },
});

export { Header };

export default withTheme<HeaderProps, {}>(Header, 'Header');
