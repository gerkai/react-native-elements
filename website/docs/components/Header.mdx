---
id: header

title: Header
---

import Usage from './usage/Header/Header.mdx';

import { Header } from 'react-native-elements';

import Tabs from '@theme/Tabs';

import TabItem from '@theme/TabItem';

Headers are navigation components that display information and actions relating to the current screen.

**Note:**

Make sure that you have completed [Step 3](/docs/installation.md#install-react-native-safe-area-context) in the setup guide before using `Header`.

## Usage

<Usage />

## Props

### Header

- [ViewComponent](#viewcomponent)
- [backgroundColor](#backgroundcolor)
- [backgroundImage](#backgroundimage)
- [backgroundImageStyle](#backgroundimagestyle)
- [barStyle](#barstyle)
- [centerComponent](#centercomponent)
- [centerContainerStyle](#centercontainerstyle)
- [children](#children)
- [containerStyle](#containerstyle)
- [elevated](#elevated)
- [leftComponent](#leftcomponent)
- [leftContainerStyle](#leftcontainerstyle)
- [linearGradientProps](#lineargradientprops)
- [placement](#placement)
- [rightComponent](#rightcomponent)
- [rightContainerStyle](#rightcontainerstyle)
- [statusBarProps](#statusbarprops)

## Reference

### Header

#### ViewComponent

Component for container.

| Type            | Default                           |
| --------------- | --------------------------------- |
| React Component | ImageBackground or View Component |

---

#### backgroundColor

Sets backgroundColor of the parent component.

| Type   | Default |
| ------ | ------- |
| string | None    |

---

#### backgroundImage

Sets backgroundImage for parent component.

| Type                | Default |
| ------------------- | ------- |
| ImageSourcePropType | None    |

---

#### backgroundImageStyle

Styling for backgroundImage in the main container.

| Type       | Default |
| ---------- | ------- |
| ImageStyle | None    |

---

#### barStyle

Sets the color of the status bar text.

| Type           | Default |
| -------------- | ------- |
| StatusBarStyle | None    |

---

#### centerComponent

Define your center component here.

| Type | Default |
| ---- | ------- |
| any  | None    |

---

#### centerContainerStyle

Styling for container around the centerComponent.

| Type               | Default |
| ------------------ | ------- |
| View style(Object) | None    |

---

#### children

Add children component to the header.

| Type                                 | Default |
| ------------------------------------ | ------- |
| ReactNode and (Element or Element[]) | []      |

---

#### containerStyle

Styling around the main container.

| Type               | Default |
| ------------------ | ------- |
| View style(Object) | None    |

---

#### elevated

Elevation for header

| Type    | Default |
| ------- | ------- |
| boolean | None    |

---

#### leftComponent

Define your left component here.

| Type | Default |
| ---- | ------- |
| any  | None    |

---

#### leftContainerStyle

Styling for container around the leftComponent.

| Type               | Default |
| ------------------ | ------- |
| View style(Object) | None    |

---

#### linearGradientProps

Displays a linear gradient. See [usage](#lineargradient-usage).

| Type   | Default |
| ------ | ------- |
| Object | None    |

---

#### placement

Alignment for title.

| Type                          | Default |
| ----------------------------- | ------- |
| "center" or "left" or "right" | center  |

---

#### rightComponent

Define your right component here.

| Type | Default |
| ---- | ------- |
| any  | None    |

---

#### rightContainerStyle

Styling for container around the rightComponent.

| Type               | Default |
| ------------------ | ------- |
| View style(Object) | None    |

---

#### statusBarProps

Accepts all props for StatusBar.

| Type           | Default |
| -------------- | ------- |
| StatusBarProps | None    |

---
