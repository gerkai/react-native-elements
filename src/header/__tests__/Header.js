import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Header from '../Header';

const btnCfg = { icon: 'home' };
const titleCfg = { text: 'This is a title' };

describe('Header Component', () => {
  it('should render without issues', () => {
    const component = shallow(<Header />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render left component by passing a config through props', () => {
    const component = shallow(
      <Header
        leftComponent={btnCfg}
      />
    );

    expect(component.find('NavButton').length).toBe(1);
  });

  it('should render right component by passing a config through props', () => {
    const component = shallow(
      <Header
        rightComponent={btnCfg}
      />
    );

    expect(component.find('NavButton').length).toBe(1);
  });

  it('should render center component by passing a config through props', () => {
    const component = shallow(
      <Header
        centerComponent={titleCfg}
      />
    );

    expect(component.find('Title').length).toBe(1);
  });

  it('should accept props for StatusBar', () => {
    const component = shallow(
      <Header
        statusBarProps={{ hidden: true }}
      />
    );

    expect(component.find('StatusBar').props().hidden).toBe(true);
  });
});
