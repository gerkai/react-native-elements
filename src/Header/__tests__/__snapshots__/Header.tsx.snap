// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should apply values from theme 1`] = `
<View
  replaceTheme={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "pink",
      "borderBottomColor": "#f2f2f2",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 10,
      "paddingVertical": 10,
    }
  }
  testID="headerContainer"
  updateTheme={[Function]}
>
  <RNCSafeAreaView
    edges={
      Array [
        "left",
        "top",
        "right",
      ]
    }
    style={
      Object {
        "flexDirection": "row",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 3,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
        }
      }
    />
  </RNCSafeAreaView>
</View>
`;

exports[`Header Component should render center component by passing a config through props 1`] = `
<Fragment>
  <StatusBar
    animated={false}
    backgroundColor="#2089dc"
    showHideTransition="fade"
    translucent={true}
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2089dc",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
    testID="headerContainer"
  >
    <RNCSafeAreaView
      edges={
        Array [
          "left",
          "top",
          "right",
        ]
      }
      style={
        Object {
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <Children
        placement="left"
        style={
          Object {
            "flex": 1,
          }
        }
      />
      <Children
        placement="center"
        style={
          Object {
            "flex": 3,
          }
        }
      >
        <Component />
      </Children>
      <Children
        placement="right"
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </RNCSafeAreaView>
  </View>
</Fragment>
`;

exports[`Header Component should render left component by passing a component through props 1`] = `
<Fragment>
  <StatusBar
    animated={false}
    backgroundColor="#2089dc"
    showHideTransition="fade"
    translucent={true}
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2089dc",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
    testID="headerContainer"
  >
    <RNCSafeAreaView
      edges={
        Array [
          "left",
          "top",
          "right",
        ]
      }
      style={
        Object {
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <Children
        placement="left"
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Button
          onPress={[Function]}
          title="Test button"
        />
      </Children>
      <Children
        placement="center"
        style={
          Object {
            "flex": 3,
          }
        }
      />
      <Children
        placement="right"
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </RNCSafeAreaView>
  </View>
</Fragment>
`;

exports[`Header Component should render left component by passing a config through props 1`] = `
<Fragment>
  <StatusBar
    animated={false}
    backgroundColor="#2089dc"
    showHideTransition="fade"
    translucent={true}
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2089dc",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
    testID="headerContainer"
  >
    <RNCSafeAreaView
      edges={
        Array [
          "left",
          "top",
          "right",
        ]
      }
      style={
        Object {
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <Children
        placement="left"
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Component />
      </Children>
      <Children
        placement="center"
        style={
          Object {
            "flex": 3,
          }
        }
      />
      <Children
        placement="right"
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </RNCSafeAreaView>
  </View>
</Fragment>
`;

exports[`Header Component should render right component by passing a component through props 1`] = `
<Fragment>
  <StatusBar
    animated={false}
    backgroundColor="#2089dc"
    showHideTransition="fade"
    translucent={true}
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2089dc",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
    testID="headerContainer"
  >
    <RNCSafeAreaView
      edges={
        Array [
          "left",
          "top",
          "right",
        ]
      }
      style={
        Object {
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <Children
        placement="left"
        style={
          Object {
            "flex": 1,
          }
        }
      />
      <Children
        placement="center"
        style={
          Object {
            "flex": 3,
          }
        }
      />
      <Children
        placement="right"
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Button
          onPress={[Function]}
          title="Test button"
        />
      </Children>
    </RNCSafeAreaView>
  </View>
</Fragment>
`;

exports[`Header Component should render right component by passing a config through props 1`] = `
<Fragment>
  <StatusBar
    animated={false}
    backgroundColor="#2089dc"
    showHideTransition="fade"
    translucent={true}
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2089dc",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
    testID="headerContainer"
  >
    <RNCSafeAreaView
      edges={
        Array [
          "left",
          "top",
          "right",
        ]
      }
      style={
        Object {
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <Children
        placement="left"
        style={
          Object {
            "flex": 1,
          }
        }
      />
      <Children
        placement="center"
        style={
          Object {
            "flex": 3,
          }
        }
      />
      <Children
        placement="right"
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Component />
      </Children>
    </RNCSafeAreaView>
  </View>
</Fragment>
`;

exports[`Header Component should render with backgroundImage 1`] = `
<Fragment>
  <StatusBar
    animated={false}
    backgroundColor="#2089dc"
    showHideTransition="fade"
    translucent={true}
  />
  <ImageBackground
    source={
      Object {
        "uri": "http://google.com",
      }
    }
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2089dc",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
    testID="headerContainer"
  >
    <RNCSafeAreaView
      edges={
        Array [
          "left",
          "top",
          "right",
        ]
      }
      style={
        Object {
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <Children
        placement="left"
        style={
          Object {
            "flex": 1,
          }
        }
      />
      <Children
        placement="center"
        style={
          Object {
            "flex": 3,
          }
        }
      />
      <Children
        placement="right"
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </RNCSafeAreaView>
  </ImageBackground>
</Fragment>
`;

exports[`Header Component should render with backgroundImageStyle 1`] = `
<Fragment>
  <StatusBar
    animated={false}
    backgroundColor="#2089dc"
    showHideTransition="fade"
    translucent={true}
  />
  <View
    imageStyle={
      Object {
        "opacity": 0.1,
      }
    }
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2089dc",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
    testID="headerContainer"
  >
    <RNCSafeAreaView
      edges={
        Array [
          "left",
          "top",
          "right",
        ]
      }
      style={
        Object {
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <Children
        placement="left"
        style={
          Object {
            "flex": 1,
          }
        }
      />
      <Children
        placement="center"
        style={
          Object {
            "flex": 3,
          }
        }
      />
      <Children
        placement="right"
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </RNCSafeAreaView>
  </View>
</Fragment>
`;

exports[`Header Component should render without issues 1`] = `
<Fragment>
  <StatusBar
    animated={false}
    backgroundColor="#2089dc"
    showHideTransition="fade"
    translucent={true}
  />
  <ImageBackground
    source={
      Object {
        "uri": "https://cdn.pixabay.com/photo/2015/04/23/22/00/tree-736885_1280.jpg",
      }
    }
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2089dc",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
    testID="headerContainer"
  >
    <RNCSafeAreaView
      edges={
        Array [
          "left",
          "top",
          "right",
        ]
      }
      style={
        Object {
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <Children
        placement="left"
        style={
          Object {
            "flex": 1,
          }
        }
      />
      <Children
        placement="center"
        style={
          Object {
            "flex": 3,
          }
        }
      />
      <Children
        placement="right"
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </RNCSafeAreaView>
  </ImageBackground>
</Fragment>
`;
